tfx/examples/airflow_workshop/taxi/setup/setup_demo.sh

```diff
@@ -22,48 +22,27 @@ NORMAL=$(tput sgr0)
 
 printf "${GREEN}Installing TFX workshop${NORMAL}\n\n"
 
-printf "${GREEN}Refreshing setuptools to avoid _NamespacePath issues${NORMAL}\n"
-pip uninstall setuptools -y && pip install setuptools
-
-printf "${GREEN}Installing httplib2 for Beam compatibility${NORMAL}\n"
-pip install httplib2==0.12.0
-
 printf "${GREEN}Installing pendulum to avoid problem with tzlocal${NORMAL}\n"
-pip install pendulum==1.4.4
+pip install pendulum
 
 printf "${GREEN}Installing TFX${NORMAL}\n"
-pip install pyarrow==0.15.0
-pip install apache_beam==2.17.0
-pip install tfx==0.21.1
-
-printf "${GREEN}Installing Google API Client${NORMAL}\n"
-pip install google-api-python-client
-
-printf "${GREEN}Installing required Jupyter version${NORMAL}\n"
-pip install ipykernel
-ipython kernel install --user --name=tfx
-pip install --upgrade notebook==5.7.8
-jupyter nbextension install --py --symlink --sys-prefix tensorflow_model_analysis
-jupyter nbextension enable --py --sys-prefix tensorflow_model_analysis
-
-printf "${GREEN}Installing packages used by the notebooks${NORMAL}\n"
-pip install matplotlib
-pip install papermill
-pip install pandas
-pip install networkx
-
-# # Docker images
-printf "${GREEN}Installing docker${NORMAL}\n"
-pip install docker
+pip install pyarrow==5.0.0
+pip install apache_beam==2.38.0
+pip install tfx==1.9.1
```

Comment:
How about installing like... 

pip install tfx[airflow]==1.9.1 ? Then we don't need to install apache-airflow separately.

And in general, it is better to install all packages in a single command, like...

`pip install pyarrow==5.0.0 apache_beam=2.38.0 tfx[airflow]==1.9.1 ....

Because then all the dependencies will be considered at once.

Reply:
> `pip install tfx[airflow]==1.9.1`

Do we need to pin the version?

Reply:
Other versions of TFX will not work with the latest apache airflow.

Reply:
@jiyongjung0 I was following the former format of that tutorial. I think the separation in Airflow and TFX installations in the setup is for a better understanding since its a tutorial. We can change it if necessary.

Reply:
Thank you for the comment. But tfx[airflow] is the recommended installation method to avoid potential version issues. Because pip only considers versions that were specified at the current command, this is the way that minimize the trouble with version conflicts.

Reply:
I tried to use pip install tfx[airflow]. tfx[airflow] tries to install apache-airflow[mysql] and i think jupyterlab doesn't have related mysql libraries to support this installation. I got the following error.
 
```
Collecting mysqlclient>=1.3.6
  Downloading mysqlclient-2.1.1.tar.gz (88 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 88.1/88.1 kB 12.3 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [16 lines of output]
      /bin/sh: 1: mysql_config: not found
      /bin/sh: 1: mariadb_config: not found
      /bin/sh: 1: mysql_config: not found
      Traceback (most recent call last):
        File "<string>", line 36, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-1_osxu69/mysqlclient_7cf2343cc4764ef3b785a8d47b57e04b/setup.py", line 15, in <module>
          metadata, options = get_config()
        File "/tmp/pip-install-1_osxu69/mysqlclient_7cf2343cc4764ef3b785a8d47b57e04b/setup_posix.py", line 70, in get_config
          libs = mysql_config("libs")
        File "/tmp/pip-install-1_osxu69/mysqlclient_7cf2343cc4764ef3b785a8d47b57e04b/setup_posix.py", line 31, in mysql_config
          raise OSError("{} not found".format(_mysql_config_path))
      OSError: mysql_config not found
      mysql_config --version
      mariadb_config --version
      mysql_config --libs
      [end of output]
```

Reply:
Oh, right. I thought that the tutorial was using mysql. Then we cannot use [airflow] extension. Thank you for the information.

Please just change the code to install all in a single pip command like

`pip install pyarrow==5.0.0 apache_beam=2.38.0 tfx==1.9.1 ....`

Thanks!

Reply:
@jiyongjung0 Updated setup_demo.sh to install packages in single command.